# SRA_variant_search   
Search your favorite alleles in your favorite organisms in SRA database   

**Usage:**   

Copy this repository:   
```
git clone https://github.com/SichongP/SRA_variant_search
cd SRA_variant_search
```
Make a working directory:   
```
mkdir working
cd working
```
Prepare list of accessions of genomes to look for and a list of variants of interest   

An accession list is a text file with one accession number per line. Each accession number refers to a genome in SRA.   
This file can be obtained from SRA search page:   
```
1.Go to https://www.ncbi.nlm.nih.gov/sra/
2.Search and use filters on the left or advanced search to filter results(search Equus Caballus), click genome, DNA and aligned data on left
3.On the top right corner of the result list page, click on “Send to”, choose File -> RunInfo, download. An SraRunInfo.csv file should be downloaded.
```
A variant list is a text file with one variant per line. Variant should be in the following format:
```
chr:start-end
```

See sample files *[Position.csv](https://github.com/SichongP/SRA_variant_search/blob/master/Position.csv)* and *[SraRunInfo.csv](https://github.com/SichongP/SRA_variant_search/blob/master/SraRunInfo.csv)* for formatting. (Note: Only the first column in the SraRunInfo.csv file is needed.)

Run with matching breeds at end:
```
python ../fetch -f variant_list --breeds=True --out output
```

Or, run without matching breeds:
```
python ../fetch -f variant_list --out output
```

Or use a single region instead of a list:   
```
python ../fetch -r chr1:123-456 -a SraRunInfo.csv --out output
```

Note:     
-a is optional. When not supplied, fetch will look for a file names "SraRunInfo.csv" at current working directory.   
-out is optional. When not supplied, fetch will output to file "autogenerated_allele_calling_(date).csv"   
You can provide a breeds file with -b option. By default fetch reads breed information from accompanying [breeds](https://github.com/SichongP/SRA_variant_search/blob/master/breeds) file.   
--ncpus option allows parallelization of sra searching, which will greatly improve speed.

Output are two comma-separated values files (one for genotype calling and the other for breed matching), whihch can be viewed by any text editor or MS excel.
